workflow/react-project/src/components/common/Modal/ModalContent/RandomCocktail/index.jsx

```diff
@@ -0,0 +1,56 @@
+import { useSelector } from 'react-redux';
+
+import { faStar } from '@fortawesome/free-regular-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import './style.less';
+
+const RandomCocktail = () => {
+  const randomCocktailStatus = useSelector(state => state.randomCocktail?.status);
+  const randomCocktailDetails = useSelector(state => state.randomCocktail?.details)?.drinks?.[0];
+  const randomCocktail = Object.entries(randomCocktailDetails || {});
+  const cocktailName = randomCocktailDetails?.strDrink;
+  const cocktailPicture = randomCocktailDetails?.strDrinkThumb;
+  const instructions = randomCocktailDetails?.strInstructions;
+  const ingredients = randomCocktail?.filter(item => item[0].includes('strIngredient') && item[1]);
+  const quantity = randomCocktail?.filter(item => item[0].includes('strMeasure') && item[1]);
+
+  return randomCocktailStatus !== 'fulfilled' ? (
+    <div className="spinner-wrapper">
+      <div className="spinner"></div>
+    </div>
+  ) : (
```

Comment:
make spinner a separate common component and you'll can reuse it in other parts of project.
Pass `loading` prop to it.
Add animation to spinner for spinning